test/test-MetadataStore.js

```diff
@@ -125,13 +125,8 @@ exports.test_insert_twice = function*(assert) {
 
   let items = yield gMetadataStore.asyncExecuteQuery("SELECT * FROM page_metadata");
   assert.equal(items.length, 1, "Page is inserted");
-  let error = false;
-  try {
-    yield gMetadataStore.asyncInsert([metadata]);
-  } catch (e) {
-    error = true;
-  }
-  assert.ok(error, "Re-insert the same page should be rejected!");
+  yield gMetadataStore.asyncInsert([metadata]);
```

Comment:
So we expect this to not throw errors and just fail silently as an expected silent failure case.  I like it. 

Can we also add a test that checks that a different kind of exception does still throw an error?